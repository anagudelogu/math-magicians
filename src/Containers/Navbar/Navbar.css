.hamburger {
  position: fixed;
  z-index: 10;
  top: 0;
  right: 0;
  margin: 20px;
  background-color: transparent;
  border: none;
  color: white;
}

.navbar {
  position: fixed;
  height: 100vh;
  width: 70vw;
  max-width: 320px;
  background-color: var(--light-gray);
  right: 0;
  top: 0;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  padding: 10rem 0;
  transform: translateX(100%);
  transition: 0.3s ease;
}

.navbar.showMenu {
  transform: translateX(0);
}

.navbar__title {
  font-size: 1.5rem;
}

.navbar__list {
  list-style: none;
}

.navbar__item {
  margin: 15px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 40px;
}

.navbar__item a {
  width: 100%;
  font-size: 1.2rem;
  text-align: center;
  padding: 10px 20px;
  border-radius: 1rem;
}

.navbar__item .active {
  color: white;
  background-color: var(--dark-orange);
}

@media screen and (min-width: 768px) {
  .hamburger {
    display: none;
  }

  .navbar {
    transform: translateX(0);
    height: 80px;
    width: 100%;
    max-width: 100vw;
    padding: 10px 50px;
    flex-direction: row;
    justify-content: space-between;
  }

  .navbar__title {
    font-size: 2rem;
  }

  .navbar__list {
    display: flex;
    gap: 1rem;
  }
}
