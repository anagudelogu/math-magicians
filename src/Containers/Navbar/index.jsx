import { NavLink } from 'react-router-dom';
import { useState } from 'react';
import './Navbar.css';
import { FaBars } from 'react-icons/fa';
import { MdOutlineClose } from 'react-icons/md';

const Navbar = () => {
  const [openMenu, setOpenMenu] = useState(false);

  const pages = [
    {
      id: 1,
      name: 'Home',
      route: '/',
    },
    {
      id: 2,
      name: 'Calculator',
      route: '/calculator',
    },
    {
      id: 3,
      name: 'Quote',
      route: '/quote',
    },
  ];

  const handleToggle = () => {
    setOpenMenu((prev) => !prev);
  };

  const closeMenu = () => {
    setOpenMenu(false);
  };

  const iconStyle = {
    width: '30px',
    height: '30px',
    cursor: 'pointer',
  };

  return (
    <nav className="nav__container">
      <button
        type="button"
        className="hamburger"
        onClick={handleToggle}
      >
        {openMenu ? (
          <MdOutlineClose style={{ ...iconStyle, color: 'black' }} />
        ) : (
          <FaBars style={{ ...iconStyle, color: 'white' }} />
        )}
      </button>
      <div className={`navbar ${openMenu ? 'showMenu' : ''}`}>
        <h1 className="navbar__title"> Math Magicians</h1>
        <ul className="navbar__list">
          {pages.map((page) => (
            <li className="navbar__item" key={page.id}>
              <NavLink
                className={({ isActive }) => (isActive ? 'active' : null)}
                to={page.route}
                onClick={closeMenu}
              >
                {page.name}
              </NavLink>
            </li>
          ))}
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
