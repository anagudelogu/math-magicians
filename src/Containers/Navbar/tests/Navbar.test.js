import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../index';

describe('Navigation bar', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
