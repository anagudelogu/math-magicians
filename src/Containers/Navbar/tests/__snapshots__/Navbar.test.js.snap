// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar Renders correctly 1`] = `
<nav
  className="nav__container"
>
  <button
    className="hamburger"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "white",
          "cursor": "pointer",
          "height": "30px",
          "width": "30px",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
  </button>
  <div
    className="navbar "
  >
    <h1
      className="navbar__title"
    >
       Math Magicians
    </h1>
    <ul
      className="navbar__list"
    >
      <li
        className="navbar__item"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navbar__item"
      >
        <a
          className={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navbar__item"
      >
        <a
          className={null}
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
